vxquery-core/src/main/java/org/apache/vxquery/serializer/XMLSerializer.java

```diff
@@ -346,6 +358,100 @@ private void printAttributeNode(PrintStream ps, TaggedValuePointable tvp) {
         }
     }
 
+    private void printObject(PrintStream ps, TaggedValuePointable tvp) {
+        ObjectPointable op = pp.takeOne(ObjectPointable.class);
+        TaggedValuePointable keys = pp.takeOne(TaggedValuePointable.class);
+        tvp.getValue(op);
+        try {
+            op.getKeys(keys);
+            printPair(ps, keys, op);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            pp.giveBack(keys);
+        }
+    }
+
+    private void printPair(PrintStream ps, TaggedValuePointable keys, ObjectPointable op) {
+        SequencePointable seqp = pp.takeOne(SequencePointable.class);
+        VoidPointable vp = pp.takeOne(VoidPointable.class);
+        UTF8StringPointable utf8sp = pp.takeOne(UTF8StringPointable.class);
+        TaggedValuePointable tvp = pp.takeOne(TaggedValuePointable.class);
+        try {
+            if (keys.getTag() == ValueTag.SEQUENCE_TAG) {
+                keys.getValue(seqp);
+                int len = seqp.getEntryCount();
+                ps.append('{');
+                for (int i = 0; i < len; i++) {
+                    ps.append('\"');
+                    seqp.getEntry(i, tvp);
+                    print(tvp.getByteArray(), tvp.getStartOffset(), tvp.getLength(), ps);
```

Comment:
How about create a printQuotedString function? It seems you add quotes many places. This could make this a simpler function.